repository/src/test/java/org/alfresco/opencmis/OpenCmisLocalTest.java

```diff
@@ -501,14 +505,47 @@ public void testCancelCheckoutWhileInCheckedOutState()
         NodeRef doc1WorkingCopy = cociService.getWorkingCopy(doc1NodeRef);
 
         /* Cancel Checkout */
-        cociService.cancelCheckout(doc1WorkingCopy);
+        waitForMethodToFinish(of(100, MILLIS), () ->
+                cociService.cancelCheckout(doc1WorkingCopy));
 
         /* Check if both the working copy and the document were deleted */
         NodeService nodeService = serviceRegistry.getNodeService();
         assertFalse(nodeService.exists(doc1NodeRef));
         assertFalse(nodeService.exists(doc1WorkingCopy));
     }
 
+    private void waitForMethodToFinish(Duration timeout, Runnable method)
+    {
+        final long lastStep = 10;
+        final long delayMillis = timeout.toMillis() > lastStep ? timeout.toMillis() / lastStep : 1;
+
+        for (int step = 0; step <= lastStep; step++)
+        {
+            try
+            {
+                method.run();
+                return;
+            }
+            catch (Exception e)
+            {
+                if (step == lastStep)
+                {
+                    //Method failed - no more waiting.
+                    throw e;
+                }
+            }
+        }
+        try
+        {
+            Thread.sleep(delayMillis);
+        }
+        catch (InterruptedException e)

```

Comment:
The `waitForMethodToFinish` is directly invoked from a test method. It might be better to just propagate the `InterruptedException`